The image does not work on Apple silicon

The following error occurs when trying to build on Apple silicon:

```
0.213 FAILED: shadercross 
0.213 : && /usr/bin/cc  -Wl,--no-undefined CMakeFiles/shadercross.dir/src/cli.c.o -o shadercross  -Wl,-rpath,/usr/local/lib:  libSDL3_shadercross.a  /usr/local/lib/libSDL3.so.0.2.4  /usr/local/lib/libspirv-cross-c-shared.so.0.64.0  /usr/local/lib/libdxcompiler.so && :
0.213 /usr/bin/ld: /usr/local/lib/libdxcompiler.so: error adding symbols: file in wrong format
0.213 collect2: error: ld returned 1 exit status
0.213 ninja: build stopped: subcommand failed.
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -1,6 +1,9 @@
 # The releases and commits are up to date as of 2025-03-01.
 
-FROM ubuntu:noble-20250127
+# The statement --platform=linux/amd64 causes Docker to output a warning but
+# trying to build for any other platform causes a failure on macOS (silicon)
+# since libdxil.so and libdxcompiler.so is built for x86_64.
+FROM --platform=linux/amd64 ubuntu:noble-20250127
 
 ENV DEBIAN_FRONTEND=noninteractive
 

--- a/Makefile
+++ b/Makefile
@@ -3,14 +3,19 @@
 mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
 mkfile_dir := $(abspath $(dir $(mkfile_path)))
 
+# The platform is explicitly set to linux/amd64 for reasons specified in
+# Dockerfile.
+
 docker-build:
 	docker build \
+		--platform=linux/amd64 \
 		-t shadercross \
 		.
 
 # Compiles an example shader.
 docker-run-example: docker-build
 	docker run \
+		--platform=linux/amd64 \
 		-v ${mkfile_dir}/in:/in \
 		-v ${mkfile_dir}/out:/out \
 		--rm \

--- a/Makefile
+++ b/Makefile
@@ -24,3 +24,7 @@ docker-run-example: docker-build
 			shadercross /in/triangle.vert.hlsl -o /out/triangle.vert.spv && \
 			shadercross /in/triangle.vert.hlsl -o /out/triangle.vert.msl && \
 			shadercross /in/triangle.vert.hlsl -o /out/triangle.vert.dxil"
+
+# Removes folder "out".
+clean-out:
+	rm -rf out

--- a/README.md
+++ b/README.md
@@ -6,4 +6,4 @@ is available.
 Run `make docker-run-example` to compile an example shader.
 The results can be found in `out`.
 
-The image has been tested on macOS (silicon).
+The image has been tested on Linux, macOS (silicon).

--- a/README.md
+++ b/README.md
@@ -5,3 +5,5 @@ is available.
 
 Run `make docker-run-example` to compile an example shader.
 The results can be found in `out`.
+
+The image has been tested on macOS (silicon).

--- a/README.md
+++ b/README.md
@@ -6,4 +6,4 @@ is available.
 Run `make docker-run-example` to compile an example shader.
 The results can be found in `out`.
 
-The image has been tested on Linux, macOS (silicon).
+The image has been tested on Linux, macOS (silicon) and Windows.

